src/character.cpp

```diff
@@ -692,7 +692,9 @@ void Character::recalc_sight_limits()
     if( has_nv() ) {
         vision_mode_cache.set( NV_GOGGLES );
     }
-    if( has_active_mutation( trait_NIGHTVISION3 ) || is_wearing( "rm13_armor_on" ) ) {
+    if( has_active_mutation( trait_NIGHTVISION3 ) || is_wearing( "rm13_armor_on" ) ||
+        ( has_effect( effect_riding ) && g->u.mounted_creature &&
+          g->u.mounted_creature.get()->has_flag( MF_MECH_RECON_VISION ) ) ) {
```

Comment:
This condition (effect, mounted_creature, has_flag)  is repeated over and over again and should be moved into a function.